src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -1462,16 +1462,15 @@ void TurboAssembler::li(Register rd, Operand j, LiFlags mode) {
     }
 
     RecordRelocInfo(j.rmode(), immediate);
-    // FIXME(RISC_V): Does this case need to be constant size?
-    li_constant(rd, immediate);
+    li_ptr(rd, immediate);
```

Comment:
Is this case always an address? Do we need to check the mode?

Reply:
Yes we can delete  the check and if you insist I will do it.

Reply:
Also RecordRelocInfo(...) is called in one case and not in the else part.